azurelinuxagent/common/utils/networkutil.py

```diff
@@ -226,43 +210,62 @@ def __execute_cmd(cmd):
             raise Exception(msg)
 
     @staticmethod
-    def add_iptables_rules(wait, dst_ip, uid):
-        # The order in which the below rules are added matters for the ip table rules to work as expected
-        AddFirewallRules.__raise_if_empty(dst_ip, "Destination IP")
-        AddFirewallRules.__raise_if_empty(uid, "User ID")
-
-        accept_tcp_rule = AddFirewallRules.get_accept_tcp_rule(AddFirewallRules.APPEND_COMMAND, dst_ip, wait=wait)
-        AddFirewallRules.__execute_cmd(accept_tcp_rule)
+    def __execute_check_command(cmd):
+        # Here we primarily check if an  iptable rule exist. True if it exits , false if not
+        try:
+            shellutil.run_command(cmd)
+            return True
+        except CommandError as err:
+            # return code 1 is expected while using the check command. Raise if encounter any other return code
+            if err.returncode != 1:
+                raise
+        return False
 
-        accept_rule = AddFirewallRules.get_iptables_accept_command(wait, AddFirewallRules.APPEND_COMMAND, dst_ip, uid)
-        AddFirewallRules.__execute_cmd(accept_rule)
+    @staticmethod
+    def verify_iptables_rules_exist(wait, dst_ip, uid):
+        check_cmd_tcp_rule = AddFirewallRules.get_accept_tcp_rule(AddFirewallRules.CHECK_COMMAND, dst_ip, wait=wait)
+        check_cmd_accept_rule = AddFirewallRules.get_accept_command(AddFirewallRules.CHECK_COMMAND, dst_ip, uid,
+                                                                    wait=wait)
+        check_cmd_drop_rule = AddFirewallRules.get_drop_command(AddFirewallRules.CHECK_COMMAND, dst_ip, wait=wait)
 
-        drop_rule = AddFirewallRules.get_iptables_drop_command(wait, AddFirewallRules.APPEND_COMMAND, dst_ip)
-        AddFirewallRules.__execute_cmd(drop_rule)
+        return AddFirewallRules.__execute_check_command(check_cmd_tcp_rule) and AddFirewallRules.__execute_check_command(check_cmd_accept_rule) \
+               and AddFirewallRules.__execute_check_command(check_cmd_drop_rule)
 
     @staticmethod
-    def __execute_firewalld_commands(command, dst_ip, uid):
+    def __execute_firewall_commands(dst_ip, uid, command=APPEND_COMMAND, firewalld_command="", wait=""):
+        # The order in which the below rules are added matters for the ip table rules to work as expected
+
         AddFirewallRules.__raise_if_empty(dst_ip, "Destination IP")
         AddFirewallRules.__raise_if_empty(uid, "User ID")
 
         # Firewalld.service fails if we set `-w` in the iptables command, so not adding it at all for firewalld commands
```

Comment:
This comment is not relevant here anymore, it should be added in the place we fetch the `firewalld` base command (which is divided in the 3 `get_*_rules`